components/core/src/clp_s/ArchiveReader.cpp

```diff
@@ -1,15 +1,23 @@
 #include "ArchiveReader.hpp"
 
+#include <filesystem>
+#include <string_view>
+
 #include "archive_constants.hpp"
 #include "ReaderUtils.hpp"
 
+using std::string_view;
+
 namespace clp_s {
-void ArchiveReader::open(std::string const& archive_path) {
+void ArchiveReader::open(string_view archives_dir, string_view archive_id) {
     if (m_is_open) {
         throw OperationFailed(ErrorCodeNotReady, __FILENAME__, __LINE__);
     }
     m_is_open = true;
-    m_archive_path = archive_path;
+    m_archive_id = archive_id;
```

Comment:
If I'm not mistaken, `m_archive_id` is only used in this method, so we could make it a local variable.

Reply:
we do have this method though

https://github.com/y-scope/clp/pull/435/files/674f8b414292b0bc6e94f1a7ba0e485efa134663#diff-aab9a16ec680184cc79784526e0e20dd1b8d0ad18737eda5f9a31dc1ee7bc415R97

Reply:
Sorry, I mean `m_archive_path`.